src/Conversion/ONNXToTOSA/DialectBuilder.cpp

```diff
@@ -178,8 +182,12 @@ Value TosaBuilder::mul(Value &lhs, Value &rhs, int32_t shift) {
   Type newValueType = RankedTensorType::get(
       llvm::SmallVector<int64_t, 4>(lhsType.getRank(), ShapedType::kDynamic),
       lhsType.getElementType());
+
+  auto int8Type = rewriter().getI8Type();
+  auto shiftValue = TosaBuilder::createConst(
+      ArrayRef<int8_t>{static_cast<int8_t>(shift)}, {1}, int8Type);
```

Comment:
IMO its better to change the `TosaBuilder::mul` parameter `shift` to int8_t isntead of having this hidden cast inside the function

Reply:
I mean the function signature: `Value TosaBuilder::mul(Value &lhs, Value &rhs, int32_t shift)`.
Right now it takes an `int32_t` paramter and internally cast it to `int_8`.
I would prefer it to change the function signature to `Value TosaBuilder::mul(Value &lhs, Value &rhs, int8_t shift)` so that it is clear for callers that it only supports int8 shift values